.github/workflows/main.yml

```diff
@@ -0,0 +1,68 @@
+name: goreleaser
+
+on:
+  push:
+    # not not consider simplec commit
+    branches:
+      - '!*'
+    # consider only release and pre-release tags
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: set up go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+
+      - name: cache go modules
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
```

Comment:
now builtin setup-go if I'm not mistaken

Reply:
I confirm: setup-go@v4 enables caching by default (see [here](https://github.com/marketplace/actions/setup-go-environment))
![image](https://user-images.githubusercontent.com/1496191/229704042-d5946594-2fca-413d-bc91-4f22b4b29a4c.png)